sql/parser/eval.go

```diff
@@ -547,10 +540,12 @@ type EvalContext struct {
 	NodeID        uint32
 	StmtTimestamp DTimestamp
 	TxnTimestamp  DTimestamp
+	ReCache       *RegexpCache
 	GetLocation   func() (*time.Location, error)
 }
 
 var defaultContext = EvalContext{
+	ReCache: NewRegexpCache(128), // TODO(nvanbenschoten) how should this be sized?
```

Comment:
Random data point: Python's `re` module uses a 512-entry cache. It was changed relatively recently from 100. We saw performance problems at Dropbox attributable to the 100-entry limit; I think we patched it there to 1000. 

I think 512 entries is reasonable for us; I wouldn't bother making it configurable.